scripts/build_a_package.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/sh
+
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script builds wheels for a single package when trigged from a push to
+# a tag as part of a GitHub workflow (See .github/publish-a-package.yml). The
+# wheel is then published to PyPI.
+
+set -ev
+
+if [[ -z $GITHUB_REF ]]; then
+  echo 'Failed to run script, missing workflow env variable GITHUB_REF'
+  exit -1
+fi
+
+# Follows PEP 440 versioning conventions: https://www.python.org/dev/peps/pep-0440/#pre-releases
+# Only allow publishing packages that are 1.0+ using this method
+if ! [[ $GITHUB_REF =~ ^refs/tags/(opentelemetry-.*)==([1-9]([0-9]+)?\.[0-9]+((a|b|rc)[0-9]+)?(\.(post)?[0-9]+)?)$ ]]; then
```

Comment:
I think something like this is a lot simpler if you don't care about validating against PEP440:

```suggestion
if ! [[ $GITHUB_REF =~ ^refs/tags/(opentelemetry-.*)==([1-9][0-9]*\..+)$ ]]; then
```

If you want to validate it, the regex might have bugs. There is a nice parser for this available in python [`packaging` package](https://packaging.pypa.io/en/latest/version.html). You might consider doing a `python -c` oneliner if you want to validate

Reply:
Nice I really like the `packaging` package solution! It worked on my repo's workflow too so I'll keep it like that 👍 The only thing is we don't have a custom error message now, we just throw the python traceback instead.

See this workflow that fails on [invalid version in the tag](https://github.com/NathanielRN/opentelemetry-python-contrib-1/runs/3127327199?check_suite_focus=true)